cmd/snap/cmd_run.go

```diff
@@ -67,7 +67,7 @@ func (x *cmdRun) Execute(args []string) error {
 	if x.Hook != "" && x.Command != "" {
 		return fmt.Errorf("cannot use --hook and --command together")
 	}
-	if x.Revision != "" && x.Hook == "" {
+	if x.Revision != "unset" && x.Revision != "" && x.Hook == "" {
```

Comment:
notice also that during installation of a snap itself we don't have an exposed concept of current until very late (but we know revisions)